cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -594,83 +577,122 @@ forgeTracer
-notifyBlockForging bcCounters tr = Tracer $ \case
-  Consensus.TraceForgedBlock {} -> do
-    updatedBlocksForged <- atomicModifyIORef' bcCounters (\cnts -> let nc = bcBlocksForgedNum cnts + 1
-                                                                   in (cnts { bcBlocksForgedNum = nc }, nc)
-                                                         )
-    traceCounter "blocksForgedNum" updatedBlocksForged tr
-  Consensus.TraceNodeCannotForge {} -> do
+notifyBlockForging fStats tr = Tracer $ \case
+  Consensus.TraceStartLeadershipCheck slot -> do
+    void $ mapForgingCurrentThreadStats slot fStats $
+      \fts -> (fts { ftsLeadershipChecksInitiated = ftsLeadershipChecksInitiated fts + 1 },
+               ftsLeadershipChecksInitiated fts + 1)
+  Consensus.TraceNodeCannotForge slot _ -> do
     -- It means that node have tried to forge new block, but because of misconfiguration
     -- (for example, invalid key) it's impossible.
-    updatedNodeCannotForge <- atomicModifyIORef' bcCounters $ \cnts ->
-      let nc = bcNodeCannotForgeNum cnts + 1
-      in (cnts { bcNodeCannotForgeNum = nc }, nc)
-    traceCounter "nodeCannotForge" updatedNodeCannotForge tr
-  -- The rest of the constructors.
-  _ -> pure ()
-
-notifySlotsMissedIfNeeded
-  :: IORef BlockchainCounters
-  -> Trace IO Text
-  -> Tracer IO (Consensus.TraceForgeEvent blk)
-notifySlotsMissedIfNeeded bcCounters tr = Tracer $ \case
-  Consensus.TraceNodeIsLeader {} -> do
-    updatedNodeIsLeaderNum <- atomicModifyIORef' bcCounters (\cnts -> let nc = bcNodeIsLeaderNum cnts + 1
-                                                                      in (cnts { bcNodeIsLeaderNum = nc }, nc)
-                                                            )
-    traceCounter "nodeIsLeaderNum" updatedNodeIsLeaderNum tr
-  Consensus.TraceNodeNotLeader {} -> do
-    -- Not is not a leader again, so now the number of blocks forged by this node
-    -- should be equal to the number of slots when this node was a leader.
-    counters <- readIORef bcCounters
-    let howManyBlocksWereForged = bcBlocksForgedNum counters
-        timesNodeWasALeader = bcNodeIsLeaderNum counters
-        numberOfMissedSlots = timesNodeWasALeader - howManyBlocksWereForged
-    if numberOfMissedSlots > 0
-    then do
-        -- Node was a leader more times than the number of forged blocks,
-        -- it means that some slots were missed.
-      updatesSlotsMissed <- atomicModifyIORef' bcCounters (\cnts -> let nc = bcSlotsMissedNum cnts + numberOfMissedSlots
-                                                                    in (cnts { bcSlotsMissedNum = nc }, nc)
-                                                          )
-      traceCounter "slotsMissedNum" updatesSlotsMissed tr
-    else return ()
-  -- The rest of the constructors.
+    void $ mapForgingCurrentThreadStats slot fStats $
+      \fts -> (fts { ftsNodeCannotForgeNum = ftsNodeCannotForgeNum fts + 1 },
+               ftsNodeCannotForgeNum fts + 1)
```

Comment:
You're not using the return value of `mapForgingCurrentThreadStats`, so I wouldn't bother with `ftsNodeCannotForgeNum fts + 1` which is then thrown away by `void`. I would even consider writing+using a variant of `mapForgingCurrentThreadStats(_)` which just takes an update function and doesn't return an additional value.

Reply:
Oh, right, that's an artifact of refactoring and forgetting it's no longer used..